x-pack/plugins/timelines/server/search_strategy/index_fields/index.ts

```diff
@@ -136,18 +136,20 @@ export const requestIndexFieldSearch = async (
       []
     );
     if (!request.onlyCheckIfIndicesExist) {
-      const fieldDescriptor = await Promise.all(
-        indicesExist.map(async (index, n) => {
-          if (index.startsWith('.alerts-observability')) {
-            return indexPatternsFetcherAsInternalUser.getFieldsForWildcard({
+      const fieldDescriptor = (
+        await Promise.all(
+          indicesExist.map(async (index, n) => {
+            if (index.startsWith('.alerts-observability')) {
+              return indexPatternsFetcherAsInternalUser.getFieldsForWildcard({
+                pattern: index,
+              });
+            }
+            return indexPatternsFetcherAsCurrentUser.getFieldsForWildcard({
               pattern: index,
             });
-          }
-          return indexPatternsFetcherAsCurrentUser.getFieldsForWildcard({
-            pattern: index,
-          });
-        })
-      );
+          })
+        )
+      ).map((response) => response.fields || []);
```

Comment:
Wdy about instead of mapping here again, to move the `fields` accessor to the previous `map`:

```
const { fields } = await indexPatternsFetcherAsCurrentUser.getFieldsForWildcard({
  pattern: index,
});
return fields || [];
```

Reply:
seems it would be bit more complicated due to the if condition in first map, but yeah it could be done, end result would be the same. Do you feel this should be addressed ?

Reply:
Ah yes, it would need to be duplicated basically...hmm..I don't think it needs to be changed in that case